### PR TITLE
Make tests suite

### PR DESCRIPTION
**Pull request by kristyhoran**
- Added in a test data set of 4 listeria sequences
- `bohra test` downloads the data and references 
- `bohra test` runs the full bohra pipeline

### CODE REVIEW

File: bohra/SnpDetection.py [hunk 1]
```diff
@@ -778,14 +779,29 @@ class TestBohra(SetupInputFiles):
 
     def __init__(self,args):
 
-        self.read_path = args.read_path
+        self.read_path = f"{pathlib.Path.cwd() / 'test_data'}"
+        self.isolate_list = ['ERR1102348','ERR1102353','ERR1102355','ERR1102356']
+        self.download_stub = "https://github.com/MDU-PHL/bohra/tree/make_tests_suite/bohra/tests/data/"
```
Comment (stroehleina) [inline bohra/SnpDetection.py]: This path will be dead once `make_test_suite` is merged and you delete the branch, should `master` be reliant on existence of this dev branch?

**Comment by Wytamma** [comment]
The biggest thing I can see is that it looks like the tests/test data are nested within the package? I think it will download all the test data at install time? e.g. running `pip install install https://github.com/MDU-PHL/bohra/archive/make_tests_suite.zip` downloads 350MB of data. Or is that stripped out during the wheel building? 

Typically you'd keep tests at the top level of the repo and run the as part of CI/during development. Including them as checks to confirm installation is also valid but you should probably move the test data to the top directory so it's only not packaged with the library (assuming that is happening).

**Review by stroehleina — COMMENTED** [review]
Looks good, few comments.

1. `MANIFEST.in` of `make_test_suite` seems to be behind `master` missing `include bohra/bohra_install.sh` see [here](https://github.com/MDU-PHL/bohra/compare/make_tests_suite...master#diff-41d5a52589e0480be9c099d2bba7a8135b8b0d71bcbb8df3582a8df1c2295003)

2. Hard-coded link to `make_test_suite` branch in `SnpDetection.py`, see comment

3. Running test seems to not create `test-results` folder which gives errors in CI steps `Uploading test results` and `Uploading artifacts` see [here](https://app.circleci.com/pipelines/github/MDU-PHL/bohra/273/workflows/e77ff165-662b-4c54-90f2-79097e8e4dda/jobs/238/parallel-runs/0/steps/0-108) and [here](https://app.circleci.com/pipelines/github/MDU-PHL/bohra/273/workflows/e77ff165-662b-4c54-90f2-79097e8e4dda/jobs/238/parallel-runs/0/steps/0-109). Not sure if this is an issue, maybe you have to specifically create the `test-results` folder?

File: bohra/SnpDetection.py (hunk 1 above)
Comment (stroehleina) [reply bohra/SnpDetection.py]: https://github.com/MDU-PHL/bohra/blob/98e390a15d246c011974043cf3dfbe8b94027f69/bohra/SnpDetection.py#L784

Comment (stroehleina) [reply bohra/SnpDetection.py]: I think you can just use a [permalink](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files) instead, which should remain even after deleting `make_test_suite`:

https://github.com/MDU-PHL/bohra/tree/98e390a15d246c011974043cf3dfbe8b94027f69/bohra/tests/data

**Comment by kristyhoran** [comment]
Thanks @Wytamma and @stroehleina  I will make changes in another pull request

**Pull request closed without merging**